go/vt/topo/memorytopo/memorytopo.go

```diff
@@ -70,6 +70,9 @@ type Factory struct {
 	// err is used for testing purposes to force queries / watches
 	// to return the given error
 	err error
+	// listErr is used for testing purposed to fake errors from
+	// calls to List.
+	listErr error
```

Comment:
It looks like err is meant to be used in the same way. Any reason not to use that for list errors as well? Fine to have the additional one if needed or preferred for some reason — but it feels like we might be adding unnecessary code and complexity.

Reply:
We cannot use `err` because it will fail _every_ call to the topo. We want `List` to fail but the subsequent `Get` to pass. `err` was the first thing I looked at, and it doesn't work for this test case.